_source/_posts/2022-01-26-webassembly-on-kubernetes-with-rust.md

```diff
@@ -0,0 +1,365 @@
+
+```bash
+cargo build --release --target wasm32-wasi
+```
+
+That's it. You have successfully created a WebAssembly binary using Rust.
+
+## Run the workload locally (optional)
+
+Let's run the workload locally using [wasmtime](https://wasmtime.dev), a small JIT-style runtime for Wasm and WASI. Since wasmtime doesn't support networking out of the box, we need to use the [wrapper](https://github.com/deislabs/wasi-experimental-http#testing-using-the-wasmtime-http-binary) provided by [wasi-experimental-http](https://github.com/deislabs/wasi-experimental-http). You can build it from source it using the below command.
+
+```bash
+git clone https://github.com/deislabs/wasi-experimental-http.git
+# Build for your platform
+cargo build
+# move to any location that is added to your PATH variable
+mv ./target/debug/wasmtime-http ~/bin/wasmtime-http
+```
+
+Now run the below command from the `rust-wasm` project folder:
+
+```bash
+wasmtime-http target/wasm32-wasi/release/rust-wasm.wasm -a https://catfact.ninja/fact -e RUST_LOG=info
+```
+
+## Run the workload in Kubernetes
+
+Before you can run the workload in Kubernetes, you need to push the binary to a registry that supports OCI artifacts. OCI complaint registries can be used for any OCI artifact, including Docker images, Wasm binaries, and so on. Docker Hub currently does not support OCI artifacts; hence you can use another registry like [GitHub Package Registry](https://github.com/features/packages), [Azure Container Registry](https://azure.microsoft.com/en-in/services/container-registry/) or [Google Artifact Registry](https://cloud.google.com/artifact-registry). I'll be using GitHub Package Registry as it's the simplest to get started, and most of you might already have a GitHub account.
+
+First, you need to log in to GitHub Package Registry using `docker login`. Create a [personal access token on GitHub](https://github.com/settings/tokens) with the `write:packages` scope and use that to log in to the registry.
+
+```bash
+export CR_PAT=<your-token>
+echo $CR_PAT | docker login ghcr.io -u <Your GitHub username> --password-stdin
+```
+
+Now you need to push your Wasm binary as an OCI artifact; for this, you can use the [wasm-to-oci](https://github.com/engineerd/wasm-to-oci/releases) CLI. Use the below command to install it on your machine. Download the appropriate version for your OS.
+
+```bash
+# Install for Linux
+curl -LO https://github.com/engineerd/wasm-to-oci/releases/download/v0.1.2/linux-amd64-wasm-to-oci
+# move to any location that is added to your PATH variable
+mv linux-amd64-wasm-to-oci ~/bin/wasm-to-oci
+```
+
+Now you can push the binary you built earlier to the GitHub Package Registry. Run the below command from the `rust-wasm` folder.
+
+```bash
+wasm-to-oci push target/wasm32-wasi/release/rust-wasm.wasm ghcr.io/<your GitHub user>/rust-wasm:latest
+```
+
+You should see a successful message. Now check your GitHub packages page on your profile, and you should see the artifact listed.
```

Comment:
```suggestion
You should see a successful message. Now check the GitHub Packages page on your profile, and you should see the artifact listed.
```
no need to repeat "your" 
GitHub Packages capitalized as name of product/page UI